### DISCUSSION TITLE
Custom libraries for Czjzek distributions

### DISCUSSION THREAD

**Discussion by hbrmn** [smeerten/ssnake#124] [General]
Greetings!

In the original publication on ssNake it reads that custom libraries can be created for use in the Czjzek model, e.g. via SIMPSON, which additionally account for CSA or dipolar interactions.

Since I could not find any further instructions in the manuals or elsewhere I'd like to ask what the structure of said library is like and how we can create it.

Thanks a lot!
Henrik B.

**Comment by smeerten** [comment]
Thanks for bringing this up, I made an issue #125 to include this in the reference manual.

To provide a short answer to your question, ssnake can load files from a directory for the library when the filenames in this directory have the following structure:
`<libname>-<eta>-<Cq>.<ext>`
Where Cq is provided in MHz. ssNake will load all files which start with the same libname. For example, The following files in a directory could be loaded as a library:
- test-0.0-1.2.spe
- test-1.0-1.2.spe
- test-0.0-1.5.spe
- test-1.0-1.5.spe

This results in a grid of 2x2 with Cq limits 1.2 to 1.5 MHz and eta limits 0.0 to 1.0
